### PR TITLE
DoctorPatientAssignment Repository Layer

### PR DESCRIPTION
**Pull request by balajisandiparthi2002**
Created DAO Layer for DoctorPatientAssignment Entity.

### CODE REVIEW

File: src/main/resources/messages.properties [hunk 1]
```diff
@@ -0,0 +1,3 @@
+validation.doctorID.mandatory=doctorID is mandatory.
+validation.patientID.mandatory=patientID is mandatory.
+db.collection.name=doctor_patient_assignment
```
Comment (ashutosh-2609) [inline src/main/resources/messages.properties]: why collection name is in messages.properties
It should be in constants

File: src/main/java/com/theelixrlabs/healthcare/dao/DoctorPatientAssignmentRepository.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+@Repository
+public interface DoctorPatientAssignmentRepository extends MongoRepository<DoctorPatientAssignmentModel, UUID> {
+    /**
+     * Method declaration to find a DoctorPatientAssignmentModel by DoctorID and PatientID.
+     *
+     * @param doctorID    UUID of the doctor associated with the assignment.
+     * @param patientID    UUID of the patient associated with the assignment.
+     * @return Optional containing DoctorPatientAssignmentModel if found, empty otherwise.
+     */
+    Optional<DoctorPatientAssignmentModel> findByDoctorIDAndPatientID(UUID doctorID, UUID patientID);
```
Comment (ashutosh-2609) [inline src/main/java/com/theelixrlabs/healthcare/dao/DoctorPatientAssignmentRepository.java:21]: is doctorID and patientID combination is going to be unique across the collection ?

**Review by ashutosh-2609 — CHANGES_REQUESTED** [review]
Added minor comments

File: src/main/resources/messages.properties (hunk 1 above)
Comment (balajisandiparthi2002) [reply src/main/resources/messages.properties]: okay, changed it.

File: src/main/java/com/theelixrlabs/healthcare/dao/DoctorPatientAssignmentRepository.java (hunk 2 above)
Comment (balajisandiparthi2002) [reply src/main/java/com/theelixrlabs/healthcare/dao/DoctorPatientAssignmentRepository.java:21]: Yes, they will be unique.

File: src/main/resources/messages.properties (hunk 1 above)
Comment (EXR-85) [inline src/main/resources/messages.properties]: Give a line

**Review by tejaswinis-hub — APPROVED** [review]

File: src/main/resources/messages.properties (hunk 1 above)
Comment (balajisandiparthi2002) [inline src/main/resources/messages.properties]: Added a line.

**Review by vinodkadam22 — APPROVED** [review]

**Review by divya-sathyan — APPROVED** [review]

File: src/main/java/com/theelixrlabs/healthcare/constants/MessageConstants.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.theelixrlabs.healthcare.constants;
+
+public class MessageConstants {
+    public static final String DB_COLLECTION_NAME = "doctor_patient_assignment";
```
Comment (ashutosh-2609) [inline src/main/java/com/theelixrlabs/healthcare/constants/MessageConstants.java]: why in message constants ?
There should be different constant file for specific to these fields 
Move to PatientsConstants,java

**Review by ashutosh-2609 — CHANGES_REQUESTED** [review]
Check the latest comment and resolve the conflicts.

**Review by ashutosh-2609 — APPROVED** [review]

File: src/main/java/com/theelixrlabs/healthcare/constants/MessageConstants.java (hunk 3 above)
Comment (balajisandiparthi2002) [reply src/main/java/com/theelixrlabs/healthcare/constants/MessageConstants.java]: Resolved.

**Pull request merged**